src/linear_algebra/matrix/pos_def.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2022 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alexander Bentkamp
+-/
+import linear_algebra.matrix.spectrum
+import linear_algebra.quadratic_form.basic
+
+/-! # Positive Definite Matrices
+
+This file defines positive definite matrices and connects this notion to positive definiteness of
+quadratic forms.
+
+## Main definition
+
+ * `matrix.pos_def` : a matrix `M : matrix n n 𝕜` is positive definite if it is hermitian
+   and `xᴴMx` is greater than zero for all nonzero `x`.
+
+-/
+
+namespace matrix
+
+variables {𝕜 : Type*} [is_R_or_C 𝕜] {n : Type*} [fintype n]
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 (pi_Lp 2 (λ (_ : n), 𝕜)) _ x y
+
+/-- A matrix `M : matrix n n 𝕜` is positive definite if it is hermitian
+   and `xᴴMx` is greater than zero for all nonzero `x`. -/
+def pos_def (M : matrix n n 𝕜) :=
+M.is_hermitian ∧ ∀ x : n → 𝕜, x ≠ 0 → (0 : ℝ) < is_R_or_C.re ⟪x, M.mul_vec x⟫
```

Comment:
Can we state this with `matrix.dot_product` instead of `inner`? I think then it applies more generally.

Reply:
I used inner because it includes the conjugate transpose for the complex numbers. I could replace inner by `star` and `dot_product` if you prefer. What kind of generality are you thinking about?

Reply:
It would be nice to be able to state that a matrix of integers is positive definite without having to cast them to reals. It's not immediately clear to me how to generalize to that though

Reply:
Hm, that's difficult. We could replace `inner` by `star` and `dot_product`, and we could add a `star` instance to the integers if it doesn't exist yet. The problem is the comparison with `0` though. For the complex numbers we need to convert to the reals before we can compare. For the integers, we wouldn't want to convert to the reals.

Reply:
Strictly we don't need to convert complex numbers to reals; they actually come with an order instance that works here